### PR TITLE
fix: ref inside context

### PR DESCRIPTION
**Pull request by Varixo**

### CODE REVIEW

File: packages/qwik/src/core/shared/shared-serialization.ts [hunk 1]
```diff
@@ -577,9 +577,11 @@ type SsrNode = {
 };
 
 type DomRef = {
-  id: string;
+  ssrNode: SsrNode;
 };
 
+let isDomRef: ((obj: unknown) => obj is DomRef) | null = null;
+
```
Comment (Varixo) [inline packages/qwik/src/core/shared/shared-serialization.ts:584]: I am not sure if this is a good idea

Comment (wmertens) [reply packages/qwik/src/core/shared/shared-serialization.ts:584]: I don't see why not, although you could initially put the false function in there

File: packages/qwik/src/core/shared/shared-serialization.ts [hunk 2]
```diff
@@ -839,8 +841,8 @@ export const createSerializationContext = (
         discoveredValues.push(obj.$el$, obj.$qrl$, obj.$state$, obj.$effectDependencies$);
       } else if (isSsrNode(obj)) {
         discoveredValues.push(obj.vnodeData);
-      } else if (isDomRef(obj)) {
-        discoveredValues.push(obj.id);
+      } else if (isDomRef!(obj)) {
+        discoveredValues.push(obj.ssrNode.id);
```
Comment (wmertens) [inline packages/qwik/src/core/shared/shared-serialization.ts:845]: Btw shouldn't we rename this by using `$...$`

File: packages/qwik/src/core/shared/shared-serialization.ts [hunk 3]
```diff
@@ -1538,6 +1541,8 @@ export const canSerialize = (value: any): boolean => {
       return true;
     } else if (value instanceof Uint8Array) {
       return true;
+    } else if (isDomRef?.(value)) {
```
Comment (wmertens) [inline packages/qwik/src/core/shared/shared-serialization.ts:1544]: ... Then you don't need to ?.

**Pull request merged**